contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,356 @@
+
+1. The cluster that Bob is using is provisioned with nodes that contain one or more secondary partitions
+2. The cluster administrator runs a DaemonSet addon that discovers secondary partitions across all nodes and creates corresponding PVs for them.
+3. The addon will monitor the health of secondary partitions and taint PVs whenever the backing local storage devices becomes unhealthy.
+4. Bob creates a specialized controller (Operator) for his distributed filesystem and deploys it.
+5. The operator will identify all the nodes that it can schedule pods onto and discovers the PVs available on each of those nodes. The operator has a label selector that identifies the specific PVs that it can use (this helps preserve fast PVs for Databases for example).
+6. The operator will then create PVCs and manually bind to individual local PVs across all its nodes. 
+7. It will then create pods, manually place them on specific nodes (similar to a DaemonSet) with high enough priority and have them use all the PVCs create by the Operator on those nodes.
+8. If a pod dies, it will get replaced with a new pod that uses the same set of PVCs that the old pod had used.
+9. If a PV gets tainted as unhealthy, the Operator is expected to delete pods if they cannot tolerate device failures. 
+
+### Bob manages a specialized application that needs access to Block level storage
+
+1. The cluster that Bob uses has nodes that contain raw block devices that have not been formatted yet.
+2. The cluster admin creates a DaemonSet addon that discovers all the raw block devices on a node that are available within a specific directory and creates corresponding PVs for them with a ‘StorageType’ of ‘Block’
+```yaml
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: foo
+  annotations:
+    storage.kubernetes.io/node: k8s-node
+  labels:
+    storage.kubernetes.io/medium: hdd
+spec:
+  volume-type: local
+  storage-type: block
+  capacity:
+    storage: 100Gi
+  hostpath:
+    path: /var/lib/kubelet/storage-raw-devices/foo
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete
+```
+3. Bob creates a pod with a PVC that requests for block level access and similar to a Stateful Set scenario the scheduler will identify nodes that can satisfy the pods request.
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: myclaim
+  labels:
+    storage.kubernetes.io/medium: ssd
+spec:
+  volume-type: local
+  storage-type: block
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi
+```
+*The lifecycle of the block level PV will be similar to that of the scenarios explained earlier.* 
+
+# Open Questions & Discussion points 
+* Single vs split “limit” for storage across writable layer and logs
+* Local Persistent Volume bindings happening in the scheduler vs in PV controller
+    * Should the PV controller fold into the scheduler
```

Comment:
Yes, that is the limitation in the first phase.  We hope to solve it in the next release, but no concrete ideas yet, we're just prototyping at this point.  At a high level, the PVC binding needs to be delayed until a pod is scheduled, so that it can take into account all the other scheduling requirements of the pod.